utils/log.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func GetLogFileName(t *testing.T, label string) string {
+	fileName := strings.ReplaceAll(t.Name(), "/", "-") + "-" + label + ".log"
+	logDirectory := "logs"
+
+	err := os.MkdirAll(logDirectory, 0777)
+	if err != nil {
+		t.Fatalf("can't create log directory")
+	}
+
+	return logDirectory + "/" + fileName
+}
+
+func WriteLogFile(t *testing.T, label string, b []byte) error {
```

Comment:
It looks like all calls to this function casts a string to a byte:
`[]byte(stdout)`
Maybe we should rather just have it accept a string.

Reply:
Strings are structs pointing to a byte array. The struct gets passed by value, but not the byte array it points to. So passing a string is actually more efficient than an array, but the same as a slice, as a slice is also a struct pointing to an array.